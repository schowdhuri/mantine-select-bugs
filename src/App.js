import { NativeSelect, Select } from "@mantine/core";

function App() {
  return (
    <>
      <Select
        data={[
          { value: "react", label: "React" },
          { value: "ng", label: "Angular" },
          { value: "svelte", label: "Svelte" },
          { value: "vue", label: "Vue" },
        ]}
      />
      <NativeSelect
        data={[
          { value: "react", label: "React" },
          { value: "ng", label: "Angular" },
          { value: "svelte", label: "Svelte" },
          { value: "vue", label: "Vue" },
        ]}
      />
    </>
  );
}

export default App;
