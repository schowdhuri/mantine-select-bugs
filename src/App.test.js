import { render } from "@testing-library/react";
import App from "./App";
// Adding @emotion/jest doesn't work either
// uncomment the next two lines to try it
// import { createSerializer } from "@emotion/jest";
// expect.addSnapshotSerializer(createSerializer());

test("renders Select", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
