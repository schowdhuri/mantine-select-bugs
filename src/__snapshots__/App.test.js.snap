// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Select 1`] = `
<DocumentFragment>
  <div
    class="mantine-Select-root mantine-1m3pqry"
  >
    <div
      aria-controls="mantine-0ed4kpat5"
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-owns="mantine-0ed4kpat5-items"
      role="combobox"
      tabindex="-1"
    >
      <input
        type="hidden"
        value=""
      />
      <div
        class="mantine-Select-wrapper mantine-12sbrde"
      >
        <input
          aria-autocomplete="list"
          aria-invalid="false"
          autocomplete="nope"
          class="mantine-Select-defaultVariant mantine-Select-input mantine-1kkmtua"
          data-mantine-stop-propagation="false"
          id="mantine-0ed4kpat5"
          readonly=""
          type="text"
          value=""
        />
        <div
          class="mantine-uquezq mantine-Select-rightSection"
        >
          <svg
            data-chevron="true"
            fill="none"
            height="18"
            style="color: rgb(134, 142, 150);"
            viewBox="0 0 15 15"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mantine-NativeSelect-root mantine-1m3pqry"
  >
    <div
      class="mantine-NativeSelect-wrapper mantine-12sbrde"
    >
      <select
        aria-invalid="false"
        class="mantine-NativeSelect-defaultVariant mantine-NativeSelect-input mantine-1nymnbu"
        id="mantine-2726oaz9o"
      >
        <option
          value="react"
        >
          React
        </option>
        <option
          value="ng"
        >
          Angular
        </option>
        <option
          value="svelte"
        >
          Svelte
        </option>
        <option
          value="vue"
        >
          Vue
        </option>
      </select>
      <div
        class="mantine-uquezq mantine-NativeSelect-rightSection"
      >
        <svg
          data-chevron="true"
          fill="none"
          height="18"
          style="color: rgb(134, 142, 150);"
          viewBox="0 0 15 15"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
